doc/userDocs/usersGuide.md

```diff
@@ -259,6 +263,118 @@ but it will not stress the parts of the network stack that manage directing traf
 from external to internal addresses.  If the performance of those components is important, 
 you should use either LoadBalancer or NodePort services.
 
+#### Configuring Namespaces<a name="namespaces"></a>
+
+Weathervane runs each instance of the Auction application in a separate namespace. The 
+workload drivers also run in their own namespace.  The default names for the application 
+namespaces are __auctionw1i*n*__, where __*n*__ is the application's instance number.  The default 
+name for the workload driver namespace is __auctionw1__.  Weathervane will try 
+to create these namespaces if they don't already exist.
+
+It is possible to customize the names for the namespaces, as well as whether Weathervane 
+should try to create the namespaces, for each Kubernetes cluster specified in your 
+configuration file. Additional details about the configuration parameters discussed 
+in this section are provided in 
+[Specifying Kubernetes Clusters](#kubernetesClusters) below.
+
+
+##### Customizing the Generation of Namespace Names
+
+Weathervane generates the namespace names using three components:
+* A namespace prefix, which defaults to **auction**.
+* A unique identifier, which is __w1__ for the drivers and __w1i*n*__ for each application 
+  instance __*n*__.
+* A namespace suffix, which is blank by default.
+
+It is possible to change the strings used for the prefix and suffix using the parameters 
+`namespacePrefix` and `namespaceSuffix`.  These parameters are placed in the configuration 
+file within the definition of the Kubernetes cluster.
+
+For example, the following configuration file excerpt will use namespace names of the 
+form __appw1i*n*small__ for the cluster named *appCluster* and __driverw1small__ for 
+the cluster named *driverCluster*.
+
+```json
+"kubernetesClusters" : [ 
+  { 
+    "name" : "appCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-1",
+    "namespacePrefix" : "app",
+    "namespaceSuffix" : "small",
+  },
+  { 
+    "name" : "driverCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-2",
+    "namespacePrefix" : "driver",
+    "namespaceSuffix" : "small",
+  },
+],
+```
+
+The ability to specify a unique suffix is particularly useful when you are doing tests with different 
+configuration sizes on the same Kubernetes cluster.  Using a unique `namespaceSuffix` for 
+each configuration size will prevent the data services from being reloaded each time 
+you switch configuration size.
+
+##### Specifying Namespace Names
+
+There are situations in which namespace names must be used that do not fit the default or customized 
+name generation process.  Some examples are:
```

Comment:
It would be good to note how common these kinds of restrictions are. Some users might be wondering "Do I have these restrictions or not, and do I have to research this, or should I just try running weathervane as-is"?